source/AI.cpp

```diff
@@ -1594,7 +1581,8 @@ void AI::MoveEscort(Ship &ship, Command &command) const
 				ship.SetTargetStellar(nullptr);
 		}
 
-		if(!ship.GetTargetStellar() && !ship.GetTargetSystem())
+		// If the ship has no destination or the destination is unreachable, route to the parent's system.
+		if(!ship.GetTargetStellar() && (!ship.GetTargetSystem() || !ship.JumpFuel(ship.GetTargetSystem())))
```

Comment:
This causes ship's ability to jump to the target system be checked every frame. I haven't profiled this, but this could cause performance degradation with large number of ships. I didn't have any better ideas on how to implement this without performance impact.

Reply:
In Ship::JumpFuel's current state I'd say this is a no-go, but #7378 changes JumpFuel to be a much, much simpler calculation than it is right now, so it may be fine. An alternative would be to add a random call to this if statement so that ships only check on average some number of frames instead of every single frame.